### PR TITLE
(RSP-1794) Remove animation from Table loadMore and from ListBox

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes https://jira.corp.adobe.com/browse/RSP-1794

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/table/stories/Table.stories.tsx [hunk 1]
```diff
@@ -465,24 +467,27 @@ function AsyncLoadingExample() {
...
-              key === 'title'
-                ? <Cell textValue={item.data.title}><Link isQuiet><a href={item.data.url} target="_blank">{item.data.title}</a></Link></Cell>
-                : <Cell>{item.data[key]}</Cell>
-            }
-          </Row>)
-        }
-      </TableBody>
-    </Table>
+    <div>
+      <ActionButton onPress={() => list.remove(list.items[0].data.id)}>Remove first item</ActionButton>
```
Comment (LFDanLu) [inline packages/@react-spectrum/table/stories/Table.stories.tsx:471]: Added a button to delete first item in async list to test that loadMore animations are gone but deletion animation is still there. Pretty barebones example but can update to be more like the CRUD example later

File: packages/@react-stately/data/src/useAsyncList.ts [hunk 2]
```diff
@@ -270,7 +270,7 @@ export function useAsyncList<T, C = string>(options: AsyncListOptions<T, C>): As
     sort(sortDescriptor: SortDescriptor) {
       dispatchFetch({type: 'sorting', sortDescriptor}, sort || load);
     },
-    ...createListActions(options, fn => {
+    ...createListActions({...options, getKey}, fn => {
```
Comment (LFDanLu) [inline packages/@react-stately/data/src/useAsyncList.ts:273]: This is get the default `getKey` properly passed to `useListData` if `getKey` isn't defined by the user

**Comment by snowystinger** [comment]
lol, async loading example is so fast now

File: packages/@react-spectrum/table/src/Table.tsx [hunk 3]
```diff
@@ -238,7 +238,8 @@ function TableCollectionView({layout, collection, focusedKey, renderView, render
     onVisibleRectChange(rect) {
       bodyRef.current.scrollTop = rect.y;
       setScrollLeft(bodyRef.current, direction, rect.x);
-    }
+    },
+    transitionDuration: collection.body.props.isLoading ? 0 : 500
```
Comment (devongovett) [inline packages/@react-spectrum/table/src/Table.tsx:242]: Can we disable it only for load more? I think it looked good for initial loading... You should be able to check if the collection has items in it already to determine if it's loading more.

File: packages/@react-spectrum/table/src/Table.tsx [hunk 4]
```diff
@@ -238,7 +238,8 @@ function TableCollectionView({layout, collection, focusedKey, renderView, render
     onVisibleRectChange(rect) {
       bodyRef.current.scrollTop = rect.y;
       setScrollLeft(bodyRef.current, direction, rect.x);
-    }
+    },
+    transitionDuration: collection.body.props.isLoading && collection.body.props.items?.length > 0 ? 0 : 500
```
Comment (devongovett) [inline packages/@react-spectrum/table/src/Table.tsx:242]: This will only work with dynamic tables. You should use `collection.size` instead.

**Pull request merged**